tests/conftest.py

```diff
@@ -534,7 +581,7 @@ def fixture_s3_session() -> boto3.session.Session:
     """
 
     # start a moto server for use in testing
-    server = ThreadedMotoServer()
+    server = ThreadedMotoServer(port=5001)
```

Comment:
Thanks, great comment! I agree with this change, there's no guarantee of an open port for use. It might be nice to reuse this pattern in the future so I've added it as a descriptive fixture. With some new changes I ended up retaining endpoint_url so as to flow that data to CytoTable via `endpoint_url` for `cloudpathlib` `client(**kwargs)` specification.